java.j2seproject/src/org/netbeans/modules/java/j2seproject/J2SEActionProvider.java

```diff
@@ -703,7 +713,12 @@ private Boolean performUpdate(@NonNull final Context ctx) {
                                                 ex.getMessage()
                                             });
                                     }
-                                });
+                                };
+                                if (ctx.isAllFilesIndexing()) {
```

Comment:
Here I wonder: should the work be deferred always? For short scans, that should be only a small delay (right?) and for big scans, wouldn't it be preferable to do whatever is necessary after the scan is finished?

Reply:
The idea was that the changes will be applied as soon as possible and parallel to other scans.
The class and resource copying blocks project actions like `run`.
But I can try.

Reply:
It's not so easy. I will try in separate PR.